magda-web-client/src/Components/Dataset/New/NewDataset.tsx

```diff
@@ -55,7 +103,26 @@ export default class NewDataset extends React.Component<{}, State> {
                         filename: thisFile.name,
                         size: thisFile.size,
                         lastModified: new Date(thisFile.lastModified)
-                    });
+                    };
+
+                    // done it this way to minimise duplicate content reading/processing
+                    const input: any = {
```

Comment:
It'd be great if we could have a typescript type for `input` that we also used in the extractors  - we could even enforce it with `const extractors : (input: ExtractorInput, output: any) => void`